apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.js

```diff
@@ -17,136 +15,124 @@ import {
 } from '../../lib/editors';
 import { encodePath } from '../../lib/uri-util';
 import { getCanonicalHost } from '../../lib/url';
+import { IdSchema } from '../../lib/db-types';
 
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
-router.get('/', function (req, res, next) {
-  debug('GET /');
-  async.series(
-    [
-      (callback) => {
-        debug('query short_names');
-        sqldb.queryOneRow(sql.short_names, { course_id: res.locals.course.id }, (err, result) => {
-          if (ERR(err, callback)) return;
-          res.locals.short_names = result.rows[0].short_names;
-          callback(null);
-        });
-      },
-    ],
-    function (err) {
-      if (ERR(err, next)) return;
-      const host = getCanonicalHost(req);
-      res.locals.studentLink = new URL(
-        res.locals.plainUrlPrefix + '/course_instance/' + res.locals.course_instance.id,
-        host,
-      ).href;
-      res.locals.studentLinkQRCode = new QR({
-        content: res.locals.studentLink,
-        width: 512,
-        height: 512,
-      }).svg();
-      res.locals.infoCourseInstancePath = encodePath(
-        path.join(
-          'courseInstances',
-          res.locals.course_instance.short_name,
-          'infoCourseInstance.json',
-        ),
-      );
-      res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-    },
-  );
-});
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const short_names = await sqldb.queryRow(
+      sql.short_names,
+      { course_id: res.locals.course.id },
+      z.array(z.string()),
+    );
+    res.locals.short_names = short_names;
```

Comment:
Nit (we never use the results anywhere else locally, and splitting into two lines doesn't help readability at all.

```suggestion
    res.locals.short_names = await sqldb.queryRow(
      sql.short_names,
      { course_id: res.locals.course.id },
      z.array(z.string()),
    );
```